### PR TITLE
feat: docker updates

### PR DESCRIPTION
**Pull request by will-larkin-nhs**
## Description
- Added docker env variables
- Added an azurite setup container to create blob containers and put rules in
- Added a script to send a sample file to azurite
- Added current sample files to repo
- Added a profile for dependencies so people with macs can run docker compose  --profile dependencies up

## Context
- Docker broke
- Changes to the transform function meant that rule files have been moved to azurite

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/Set-up/azurite/rules/lookupRules.json
Comment (stephhou) [inline application/CohortManager/Set-up/azurite/rules/lookupRules.json:1]: Are we now maintaining two copies of the the rules?

File: scripts/config/gitleaks.toml [hunk 1]
```diff
@@ -26,4 +26,6 @@ paths = [
   '''application/CohortManager/spikes/AzureLocalSimulators/DTOS-2460_Event_Grid_Spike/templates/Azure EG Topic - createParticipant/template.json''',
   '''application/CohortManager/spikes/AzureLocalSimulators/DTOS-2460_Event_Grid_Spike/templates/Azure Sql Template/parameters.json''',
   '''application/CohortManager/spikes/AzureLocalSimulators/DTOS-2460_Event_Grid_Spike/templates/Azure Sql Template/template.json''',
+  '''application/CohortManager/src/Functions/CaasIntegration/receiveCaasFile/receiveCaasFile.csproj''',
```
Comment (stephhou) [inline scripts/config/gitleaks.toml:29]: Why are these changes needed in the gitleaks file?

**Review by stephhou — CHANGES_REQUESTED** [review]
Can you remove the test files? I'm not sure who this solution is aimed at, but testers and devs will probably need to add files that have specific errors in to test their code works. The would often need to provide new files anyway.

We also shouldn't have two copies of the rules files.

Also, who has requested these changes?

File: scripts/config/gitleaks.toml (hunk 1 above)
Comment (will-larkin-nhs) [reply scripts/config/gitleaks.toml:29]: False positives on Sudip's stuff

File: application/CohortManager/Set-up/azurite/rules/lookupRules.json
Comment (will-larkin-nhs) [reply application/CohortManager/Set-up/azurite/rules/lookupRules.json:1]: Will remove the ones in the validation service, transform service has not been migrated to azurite yet so I will leave those

**Comment by will-larkin-nhs** [comment]
> Can you remove the test files? I'm not sure who this solution is aimed at, but testers and devs will probably need to add files that have specific errors in to test their code works. The would often need to provide new files anyway.

I would prefer to leave them in since they are not really harming anyone and it makes things easier, but fine I will remove them.

> Also, who has requested these changes?

No-one, docker broke so needed to fix that. Then I discovered the rules had been moved to azurite so added that stuff as a productivity/ quality of life thing.

File: application/CohortManager/Set-up/azurite/rules/lookupRules.json
[the remark this answers is not in the document]
Comment (will-larkin-nhs) [reply application/CohortManager/Set-up/azurite/rules/lookupRules.json:1]: that's where the files should be now, I have removed the duplicates from the validation service

**Review by stephhou — DISMISSED** [review]

**Review by Warren-Pitterson — APPROVED** [review]

**Pull request merged**